### PR TITLE
Improving Training Performance and Scalability Documentation

### PR DESCRIPTION
**Pull request by HamzaFB**
This PR improves the docs.
A strategy for improving Memory Performance for Large Models (Billions of parameters) is PEFT.
Actual documentation does not mention it.
This PR adds PEFT and provides an example as to why this reduces memory needs.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding! 

Asking for a quick double check from @younesbelkada

File: docs/source/en/perf_train_gpu_one.md [hunk 1]
```diff
@@ -400,6 +401,25 @@ Choose which backend to use by specifying it via `torch_compile_backend` in the
...
+* Momentum: 4 bytes/param
+* Variance: 4 bytes/param
+
+Suppose a model with 7B parameters and 200 millions parameters injected with [Low Rank Adapters](https://huggingface.co/docs/peft/conceptual_guides/lora).
+
+The memory requirement for the optimizer state of the plain model would be 12 * 7 = 84 GB (assuming 7B trainable parameters).
+
+Adding Lora increases slightly the memory associated to the model weights and substantially decreases memory requirement for the optimizer state to 12 * 0.2 = 2.4GB.
+
+
```
Comment (younesbelkada) [inline docs/source/en/perf_train_gpu_one.md:421]: Read more about PEFT and its detailed usage in [the PEFT documentation](https://huggingface.co/docs/peft/) or [PEFT repository](https://github.com/huggingface/peft).

**Review by younesbelkada — APPROVED** [review]
Thanks a lot! LGTM with a single comment!

File: docs/source/en/perf_train_gpu_one.md (hunk 1 above)
Comment (younesbelkada) [inline docs/source/en/perf_train_gpu_one.md:421]:
```suggestion
Read more about PEFT and its detailed usage in [the PEFT documentation](https://huggingface.co/docs/peft/) or [PEFT repository](https://github.com/huggingface/peft).
```
Now it is better :D

Comment (younesbelkada) [reply docs/source/en/perf_train_gpu_one.md:421]: Please see: https://github.com/huggingface/transformers/pull/28497#discussion_r1452231418

**Comment by HamzaFB** [comment]
Thanks!

**Review by younesbelkada — APPROVED** [review]
Thanks again @HamzaFB !

**Pull request merged**